homeassistant/components/google/config_flow.py

```diff
@@ -41,12 +49,26 @@ def __init__(self) -> None:
         super().__init__()
         self._reauth_config_entry: config_entries.ConfigEntry | None = None
         self._device_flow: DeviceFlow | None = None
+        # We first attempt using device auth (which was recommended) with a
+        # fallback to web auth. We now prefer web auth since the credentials
```

Comment:
It's a bit unclear that we first attempt device auth even though we prefer web auth. I'd be even more explicit why we do it this way. Maybe say that Google has recommended device auth but from the Home Assistant perspective we would like to only use web auth since then we can share credentials.

Reply:
It's *slightly* different than what you said (both Google and Home Assistant recommend web auth, and historically we told users to use device auth in the integration instructions.

I attempted to rewrite this into a longer explanation to capture more context here about what is going on. This may be too long, but since its a bit subtle it may be worth preserving. 

Have a read and let me know there is anything to clarify further, or if you have corrections.

Reply:
Great!